calico.go

```diff
@@ -339,16 +367,6 @@ func cmdDel(args *skel.CmdArgs) error {
 		logger.Error(ipamErr)
 	}
 
-	calicoClient, err := CreateClient(conf)
-	if err != nil {
-		return err
-	}
-
-	ep := api.WorkloadEndpointMetadata{
-		Name:         args.IfName,
-		Node:         nodename,
-		Orchestrator: orchestrator,
-		Workload:     workload}
 	if err = calicoClient.WorkloadEndpoints().Delete(ep); err != nil {
```

Comment:
To do a atomic delete (in the sense no one else has modified it since you did your Get), you need to pass in the Metadata that was returned from the Get query - this will include revision data that etcd will use to determine whether the underlying entry has since been modified. 

So:  `calicoClient.WorkloadEndpoints().Delete(wep.Metadata)`